cirq-google/cirq_google/serialization/arg_func_langs.py

```diff
@@ -46,44 +47,21 @@
     sympy.NumberSymbol,
 )
 
-# Supported function languages in order from least to most flexible.
-# Clients should use the least flexible language they can, to make it easier
-# to gradually roll out new capabilities to clients and servers.
-LANGUAGE_ORDER = ['', 'linear', 'exp']
-
-
-def _max_lang(langs: Iterable[str]) -> str:
-    i = max((LANGUAGE_ORDER.index(e) for e in langs), default=0)
-    return LANGUAGE_ORDER[i]
-
-
-def _infer_function_language_from_circuit(value: v2.program_pb2.Circuit) -> str:
-    return _max_lang(
-        {
-            e
-            for moment in value.moments
-            for op in moment.operations
-            for e in _function_languages_from_operation(op)
-        }
-    )
-
-
-def _function_languages_from_operation(value: v2.program_pb2.Operation) -> Iterator[str]:
-    for arg in value.args.values():
-        yield from _function_languages_from_arg(arg)
-
-
-def _function_languages_from_arg(arg_proto: v2.program_pb2.Arg) -> Iterator[str]:
-    which = arg_proto.WhichOneof('arg')  # pragma: no cover
-    if which == 'func':
-        if arg_proto.func.type in ['add', 'mul']:
-            yield 'linear'
-            for a in arg_proto.func.args:
-                yield from _function_languages_from_arg(a)
-        if arg_proto.func.type in ['pow']:
-            yield 'exp'
-            for a in arg_proto.func.args:
-                yield from _function_languages_from_arg(a)
+# The sympy operations supported by the proto format
+# and their corresponding operation types
+_SUPPORTED_SYMPY_TYPE_MAPPING = {
+    sympy.Add: "add",
+    sympy.Mul: "mul",
+    sympy.Pow: "pow",
+    sympy.Equality: "==",
+    sympy.Unequality: "!=",
+    sympy.GreaterThan: ">=",
+    sympy.StrictGreaterThan: ">",
+    sympy.LessThan: "<=",
+    sympy.StrictLessThan: "<",
+}
```

Comment:
for `SympyCondition` we also need  the bitswise functions `And`, `OR`, `Not`, `Xor`, `sympy.Indexed` and `sympy.IndexedBase`

Reply:
Done.